src/java.base/share/classes/sun/security/provider/certpath/OCSP.java

```diff
@@ -214,12 +214,9 @@ public static byte[] getOCSPBytes(List<CertId> certIds, URI responderURI,
             }
 
             int contentLength = con.getContentLength();
-            if (contentLength == -1) {
-                contentLength = Integer.MAX_VALUE;
-            }
-
-            return IOUtils.readExactlyNBytes(con.getInputStream(),
-                    contentLength);
+            return (contentLength == -1) ? con.getInputStream().readAllBytes() :
```

Comment:
Well, in the case of a 404 what appears to happen is that HttpURLConnection would throw a FileNotFoundException.  That ultimately would result in a CPVE if there were no other sources of revocation information (e.g. CRL) for that certificate.

Reply:
It may be more effective/accuracy to stop read OCSP response bytes if response code is not OK.

Reply:
Logging the error code and returning with no read and not throwing an exception I believe would still work since the revocation information would be missing.  I'm wondering though if this needs to be a separate issue given that we're talking about a different use case, and one that involves the behavior of HttpURLConnection when dealing with different response codes.  I'll also check to see if there are existing tests that make CPV checks against URIs that have non-200 response codes.

Reply:
Hmmm, I was not quite correct about the HttpURLConnection behavior - it's not the 404 that's causing the issue directly, it is indeed the getContentLength when the 404 happens.  So forget a separate issue, I will deal with non-200 codes in this PR.